### ISSUE TITLE
[svelte-check] TS error when deriving from value

### ISSUE THREAD

**Issue by marekdedic** [sveltejs/language-tools#2573]
### Describe the bug

Hi,
I'm getting an error when deriving from a value caused by the fact that at the point the derived rune is used, the original variable has a fixed value:

### Reproduction

```svelte
<script lang="ts">
  let one: Array<number> | null = $state(null);
  let two = $derived(
    one === null ? 42 : one.find((x) => x > 22)
  );
</script>
```

I get `Error: Property 'find' does not exist on type 'never'.`

### Expected behaviour

No error

### System Info

- OS: Debian testing

### Which package is the issue about?

svelte-check

### Additional Information, eg. Screenshots

https://svelte.dev/playground/feba78a349b34e49aedc626604866047?version=5.1.12

Looking at the JS output, the error makes sense.

In my actual usecase, I'm binding the variable `one`.

**Comment by brunnerh** [comment]
Related to/essentially duplicate of:
- https://github.com/sveltejs/kit/issues/12831

(That issue still should be moved somewhere that isn't the SvelteKit repository.)

**Comment by marekdedic** [comment]
Yeah it is, sorry, didn't find it... So, should I close this? Or is the actual issue here in `svelte2tsx`?
